docs: update Get Started Page to Include Tasks

david-fraley opened the pull request:
If you have used AI to produce some or all of this PR, please ensure you have read our [AI Contribution guidelines](https://coder.com/docs/about/contributing/AI_CONTRIBUTING) before submitting.

This PR
* Clarifies/updates our get started page to be more clear about what's happening and provide background knowledge
* Adds in Coder Tasks as an optional step users can perform 

You can [preview the changes here](https://coder.com/docs/@david-fraley%2Fupdate-quickstart/tutorials/quickstart).

david-fraley committed:
Update quickstart docker and coder install steps

david-fraley committed:
rest of quickstart

david-fraley committed:
start of tasks docs

david-fraley committed:
continued work on tasks

david-fraley committed:
Updated tasks quickstart section

blink-so[bot] committed:
docs: fix markdown formatting issues in quickstart.md

- Fix MD037: Remove spaces in emphasis markers
- Fix MD024: Change duplicate headings to use ### instead of ##
- Fix MD009: Remove all trailing spaces
- Fix MD032: Add blank lines around lists
- Fix MD031: Add blank lines around fenced code blocks
- Fix MD022: Add blank line before heading
- Fix MD029: Fix ordered list numbering
- Fix MD047: Add proper single trailing newline

This resolves all markdownlint errors that were causing the CI build to fail.

david-fraley committed:
add back whats next

blink-so[bot] committed:
docs: fix typos and formatting in quickstart.md

- Fix typo: 'avaialable' -> 'available'
- Fix typo: 'applicaiton' -> 'application'
- Fix typo: 'ciding' -> 'coding'
- Fix typo: 'Clikc' -> 'Click'
- Remove trailing space on line 222
- Add proper single trailing newline

These fixes resolve the failing CI checks for typos and markdown formatting.

blink-so[bot] committed:
docs: format quickstart.md with prettier

- Align table columns for better readability
- Fix line wrapping for better markdown formatting

These changes resolve the fmt CI check failure.

blink-so[bot] committed:
docs: format markdown tables in quickstart.md

- Remove spaces around table separator pipes for consistent formatting
- Matches the CI's markdown-table-formatter output

This resolves the fmt CI check failure.

jcjiang reviewed (commented):
Looks good overall, just some small tweaks.

david-fraley committed:
Update tasks how-to

david-fraley committed:
add pic, remove keywords

david-fraley committed:
add note about docker running in background

david-fraley committed:
added picture for changing directories

david-fraley committed:
revert title change. it'll come later

david-fraley committed:
Merge branch 'main' into david-fraley/update-quickstart

david-fraley committed:
minor nit

david-fraley committed:
updated concepts section

david-fraley committed:
format

david-fraley committed:
more format

david-fraley committed:
Fix typo

david-fraley committed:
more formats

blink-so[bot] committed:
docs: fix trailing whitespace in quickstart.md

Remove trailing whitespace from lines in the quickstart documentation
to fix linting errors in CI.

blink-so[bot] committed:
docs: add missing newline at end of quickstart.md

The linter expects files to end with a newline character.
